### PR TITLE
Added provider of Albania 

### PR DESCRIPTION
**Pull request by notaprodev**
### What is the reason for this PR?

- [ ] A new feature
- [ ] Fixed an issue (resolve #ID)

### Author's checklist

- [ ] Follow the [Contribution Guide](https://github.com/FakerPHP/Faker/blob/main/.github/CONTRIBUTING.md)
- [ ] New features and changes are [documented](https://github.com/FakerPHP/fakerphp.github.io)

### Review checklist

- [ ] All checks have passed
- [ ] Changes are approved by maintainer

### CODE REVIEW

**Comment by pimjansen** [comment]
I see a lot of copy paste from other locales. There are docs for italian wikipedia pages?

Please add reference to sources of all of the data

**Comment by notaprodev** [comment]
thanks for the fast replay. I will fix the mentioned problems

File: src/Faker/Provider/sq_AL/Address.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+   
+    protected static $buildingNumber = ['%##', '%#', '%'];
+    protected static $streetSuffix = [
+        'Sheshi', 'Bulevardi', 'Rreth-rrotullimi', 'Kryqezimi'
+    ];
+    protected static $postcode = ['#####'];
+    protected static $state = [
+        'Belsh','Berat','Bulqizë','Cërrik','Delvinë','Devoll','Dibër','Dimal','Divjakë','Dropull','Durrës','Elbasan','Fier','Finiq','Fushë-Arrëz','Gjirokastër','Gramsh','Has','Himarë','Kamëz','Kavajë','Këlcyrë','Klos','Kolonjë','Konispol','Korçë','Krujë','Kuçovë','Kukës','Kurbin','Lezhë','Libohovë','Librazhd','Lushnjë','Malësi e Madhe','Maliq','Mallakastër','Mat','Memaliaj','Mirditë','Patos','Peqin','Përmet','Pogradec','Poliçan','Prrenjas','Pukë','Pustec','Roskovec','Rrogozhinë','Sarandë','Selenicë','Shijak','Shkodër','Skrapa','Tepelenë','Tiranë','Tropojë','Vau i Dejës','Vlorë','Vorë'
+    ];
+    protected static $country = [
```
Comment (pimjansen) [inline src/Faker/Provider/sq_AL/Address.php:16]: These are not localized

File: src/Faker/Provider/sq_AL/Address.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        'San Bartolomeo', 'Sant\'Elena', 'Saint Kitts and Nevis', 'Saint Lucia', 'Saint Martin', 'Saint Pierre and Miquelon', 'Saint Vincent and the Grenadines', 'Samoa', 'San Marino', 'Sao Tome and Principe', 'Arabia Saudita', 'Senegal', 'Serbia', 'Seychelles', 'Sierra Leone', 'Singapore', 'Slovenia', 'Isole Solomon', 'Somalia', 'Sud Africa', 'Georgia del sud e South Sandwich Islands', 'Spagna', 'Sri Lanka', 'Sudan', 'Suriname', 'Svalbard & Jan Mayen Islands', 'Swaziland', 'Svezia', 'Svizzera', 'Siria',
+        'Taiwan', 'Tajikistan', 'Tanzania', 'Tailandia', 'Timor-Leste', 'Togo', 'Tokelau', 'Tonga', 'Trinidad e Tobago', 'Tunisia', 'Turchia', 'Turkmenistan', 'Isole di Turks and Caicos', 'Tuvalu',
+        'Uganda', 'Ucraina', 'Emirati Arabi Uniti', 'Regno Unito', 'Stati Uniti d\'America', 'United States Minor Outlying Islands', 'Isole Vergini Statunitensi', 'Uruguay', 'Uzbekistan',
+        'Vanuatu', 'Venezuela', 'Vietnam',
+        'Wallis and Futuna', 'Western Sahara',
+        'Yemen',
+        'Zambia', 'Zimbabwe',
+    ];
+    protected static $cityFormats = [
+        '{{cityPrefix}} {{firstName}} {{citySuffix}}',
```
Comment (pimjansen) [inline src/Faker/Provider/sq_AL/Address.php:44]: Firstnames in a city name is odd?

File: src/Faker/Provider/sq_AL/Address.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        '{{cityPrefix}} {{firstName}}',
+        '{{firstName}} {{citySuffix}}',
+        '{{lastName}} {{citySuffix}}',
+    ];
+    protected static $streetNameFormats = [
```
Comment (pimjansen) [inline src/Faker/Provider/sq_AL/Address.php:49]: Firstnames in street names?

File: src/Faker/Provider/sq_AL/Address.php [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+     * @example 'California'
+     */
+    public static function state()
+    {
+        return static::randomElement(static::$state);
+    }
+}
+
+/*
+https://en.youbianku.com/albania
```
Comment (pimjansen) [inline src/Faker/Provider/sq_AL/Address.php:88]: This is not how we define sources, see other locales

File: src/Faker/Provider/sq_AL/Company.php [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<?php
+
+namespace Faker\Provider\sq_AL;
+
+class Company extends \Faker\Provider\Company
+{
+    protected static $formats = [
+        '{{lastName}} {{companySuffix}}',
```
Comment (pimjansen) [inline src/Faker/Provider/sq_AL/Company.php:8]: Lastnames for companies?

File: src/Faker/Provider/sq_AL/Company.php [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+        '{{lastName}} {{lastName}} {{companySuffix}}',
+        '{{lastName}}',
+        '{{lastName}}',
+    ];
+    
+    protected static $companySuffix = ['ShPK', 'ShA', 'Group'];
```
Comment (pimjansen) [inline src/Faker/Provider/sq_AL/Company.php:14]: Missing source

File: src/Faker/Provider/sq_AL/Payment.php [hunk 7]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Faker\Provider\sq_AL;
+
+class Payment extends \Faker\Provider\Payment
+{
+    /*
```
Comment (pimjansen) [inline src/Faker/Provider/sq_AL/Payment.php:7]: This is not a valid phpdoc

File: src/Faker/Provider/sq_AL/Person.php [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    protected static $femaleNameFormats = [
+        '{{firstNameFemale}} {{lastName}}',
+        '{{firstNameFemale}} {{lastName}}',
+        '{{firstNameFemale}} {{lastName}}',
+        '{{titleFemale}} {{firstNameFemale}} {{lastName}}',
+        '{{firstNameFemale}} {{lastName}}',
+        '{{titleFemale}} {{firstNameFemale}} {{lastName}}',
+    ];
+
+    protected static $firstNameMale = [
```
Comment (pimjansen) [inline src/Faker/Provider/sq_AL/Person.php:25]: No sources and wrong indent

File: src/Faker/Provider/sq_AL/Person.php [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+        'Laidin','Lavdërim','Lavdim','Lavdimir','Lazrim','Lazror','Led','Ledian','Ledion','Lek','Lind','Lir','Liridon','Lirim','Lirjon','Luan',
+        'Luftar','Luftim','Lul','Lulash','Lulëzim','Llesh','Mal','Margjel','Marin','Mat','Mentar','Mentor','Migjen','Mir','Mirash','Mirjet',
+        'Mirjon','Mirush','Ndue','Neritan','Nertil','Nertilian','Ohri','Olt','Oltian','Oltion','Orges','Darli','Bruno','Pajtim','Pëllumb',
+        'Përparim','Petrit','Pirro','Pirust','Prend','Preng','Redon','Rron','Saimir','Skënder','Skerd','Sokol','Shkëlqim','Shkëlzen','Shkëmb',
+        'Shkumbin','Kristi','Kristjan','Shpend','Shpëtim','Tanush','Taulant','Teut','Tomë','Tomor','Tonin','Trëndafil','Trim','Tritan','Ukë','Ulqin',
+        'Urim','Val','Emilian','Emiliano','Valdet','Gerjan','Gentian','Gert','Valmir','Ylber','Ylli','Zaimin','Zamir','Zef','Zymbyl',
+        
+    ];
+
+    protected static $firstNameFemale = [
```
Comment (pimjansen) [inline src/Faker/Provider/sq_AL/Person.php:46]: No sources

File: src/Faker/Provider/sq_AL/Person.php [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+        'Dhurime','Edlira','Edrina','Egzona','Elira','Elona','Emira','Emiranda','Emiriana','Emirjona','Enkela','Enkelana','Entela','Era',
+        'Eranda','Erdita','Erëzake','Erina','Erinda','Erisa','Erza','Etleva','Ëngjëllushe','Fatbardha','Fatlume','Fatmira','Fitime','Fitore',
+        'Frangë','Garentina','Gazmira','Genca','Genciana','Genta','Gentiana','Gëzime','Gonxhe','Iliresha','Jasemina','Jehona','Jehora','Jela',
+        'Jerina','Jeta','Kastriote','Luliana','Luljeta','Olta','Oltiana','Ostrala','Pajtime','Panenta','Panjëz','Paqësime','Paqësore','Valbona',
+        'Vera','Verake','Veriana','Verime','Verina','Verjona','Verore','Veroshe','Verusha','Vesa','Vilma','Vitore','Vjollca','Vjosa','Xixa',
+        'Yllëza','Zamira','Zana',
+
+    ];
+
+    protected static $lastName = [
```
Comment (pimjansen) [inline src/Faker/Provider/sq_AL/Person.php:60]: No sources

File: src/Faker/Provider/sq_AL/PhoneNumber.php [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+        '+### ### ########',
+        '+### #### #######',
+        '+### #### ########',
+        '00### ### ####',
+        '+355 0## ### ###',
+        '3## ### ###',
+        '+355 (0)## ### ###',
+    ];
+}
+/* https://en.wikipedia.org/wiki/Telephone_numbers_in_Albania
```
Comment (pimjansen) [inline src/Faker/Provider/sq_AL/PhoneNumber.php:21]: Not a valid position

**Review by pimjansen — CHANGES_REQUESTED** [review]
I added a few comments but please have a look at this locale since quality wise it is far from good.

**Comment by GlennM** [comment]
This is marked as "a new feature", but in the Contribution Guide it is mentioned that:
"_At the moment we are not looking for pull requests that add new features. We are planning for Faker 2.0's architecture._"

Are there any updates on the progress of the [new architecture](https://github.com/FakerPHP/Faker/discussions/15)?

**Comment by pimjansen** [comment]
> This is marked as "a new feature", but in the Contribution Guide it is mentioned that:
> 
> "_At the moment we are not looking for pull requests that add new features. We are planning for Faker 2.0's architecture._"
> 
> 
> 
> Are there any updates on the progress of the [new architecture](https://github.com/FakerPHP/Faker/discussions/15)?

You are right. Overall indeed we do no accept new changes that could affect this but i am not against. It only means later that all these changes are gone where the locales will be moved away from the Faker core itself.

There are still some issues at this moment on the modifiers and stacking them. If all that is there we will setup the default english language pack and start on some others from the core dev team as well.

**Comment by pimjansen** [comment]
Im going to stop the PR here. It is a new change which is not matching the contribution guide. Also the quality is far from good where we rather but the limited time we have in future development

**Comment by notaprodev** [comment]
I am sorry I marked it as new feature. Can you please point what is the problem

**Comment by bram-pkg** [comment]
Hi @notaprodev,

The main problem is that this is adding new functionality (in the form of a new provider).
Secondary, the quality and attention to detail in this PR is not up to par with the rest of our codebase. I have left you some comments on the code, to explain this further.

File: src/Faker/Provider/sq_AL/Address.php [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+        'Belsh','Berat','Bulqizë','Cërrik','Delvinë','Devoll','Dibër','Dimal','Divjakë','Dropull','Durrës','Elbasan','Fier','Finiq','Fushë-Arrëz',
+        'Gjirokastër','Gramsh','Has','Himarë','Kamëz','Kavajë','Këlcyrë','Klos','Kolonjë','Konispol','Korçë','Krujë','Kuçovë','Kukës','Kurbin',
+        'Lezhë','Libohovë','Librazhd','Lushnjë','Malësi e Madhe','Maliq','Mallakastër','Mat','Memaliaj','Mirditë','Patos','Peqin','Përmet',
+        'Pogradec','Poliçan','Prrenjas','Pukë','Pustec','Roskovec','Rrogozhinë','Sarandë','Selenicë','Shijak','Shkodër','Skrapa','Tepelenë',
+        'Tiranë','Tropojë','Vau i Dejës','Vlorë','Vorë'
+    ];
+    /**
+     * @see https://en.wikipedia.org/wiki/Category:Streets_in_Tirana
+     */
+
```
Comment (bram-pkg) [inline src/Faker/Provider/sq_AL/Address.php:20]: Spacing here?

File: src/Faker/Provider/sq_AL/Address.php [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+    protected static $street = [
+        'Tish Daija', 'Avni Rustemi', 'Jordan Misja', 'Agim Nasufi','Naim Frashri', 'Migjeni', 'Pjeter Pograni',
+        'Agim Jashari', 'Isa Boletini','Pjeter Budi', 'Fran Vasa', 'Bajram Curri', 'Azem Galica', 'Albanopoli',
+        'Egnaita', 'Lek Dukagjini', 'Lefter Talo', 'Liri Gero', 'Lidhja e Prizrenit', 'Skernderbej', 'Kupa',
+        'Haki Taha', 'Faik Konica', 'Isuf Xhelili', 'Ali Asllani', 'Fato Berberi', 'Dardania', 'Princ Vidi',
+    ];
+    /**
+     * @see https://en.wikipedia.org/wiki/Foreign_relations_of_Albania
```
Comment (bram-pkg) [inline src/Faker/Provider/sq_AL/Address.php:28]: This is a list of English country names, from English wikipedia. Not localized into Albanian.

File: src/Faker/Provider/sq_AL/Address.php [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+    ];
+    protected static $addressFormats = [
+        "{{streetSuffix}}{{street}}\n{{city}}, {{postcode}}",
+    ];
+    public static function cityName()
+    {
+        return static::randomElement(static::$city);
+    }
+
+}
```
Comment (bram-pkg) [inline src/Faker/Provider/sq_AL/Address.php:69]: No newline

File: src/Faker/Provider/sq_AL/Internet.php [hunk 15]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace Faker\Provider\sq_AL;
+
+class Internet extends \Faker\Provider\Internet
+{
+    protected static $freeEmailDomain = ['gmail.com', 'yahoo.com', 'hotmail.com'];
```
Comment (bram-pkg) [inline src/Faker/Provider/sq_AL/Internet.php:7]: No point in having these for a specific locale, either add specific Albanian domains or remove this.

File: src/Faker/Provider/sq_AL/Person.php [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+        'Gjokë','Gjon','Hekuran','Idlir','Ilir','Ilirian','Indrit','Jasemin','Jetmir','Jeton','Jetor','Jon','Kajan','Kastriot','Kelmend','Klajd',
+        'Klest','Klevis','Kodran','Krenar','Kreshnik','Krojan','Kujtim','Kushtim','Kushtrim','Labian','Laid','Laidan',
+        'Laidin','Lavdërim','Lavdim','Lavdimir','Lazrim','Lazror','Led','Ledian','Ledion','Lek','Lind','Lir','Liridon','Lirim','Lirjon','Luan',
+        'Luftar','Luftim','Lul','Lulash','Lulëzim','Llesh','Mal','Margjel','Marin','Mat','Mentar','Mentor','Migjen','Mir','Mirash','Mirjet',
+        'Mirjon','Mirush','Ndue','Neritan','Nertil','Nertilian','Ohri','Olt','Oltian','Oltion','Orges','Darli','Bruno','Pajtim','Pëllumb',
+        'Përparim','Petrit','Pirro','Pirust','Prend','Preng','Redon','Rron','Saimir','Skënder','Skerd','Sokol','Shkëlqim','Shkëlzen','Shkëmb',
+        'Shkumbin','Kristi','Kristjan','Shpend','Shpëtim','Tanush','Taulant','Teut','Tomë','Tomor','Tonin','Trëndafil','Trim','Tritan','Ukë','Ulqin',
+        'Urim','Val','Emilian','Emiliano','Valdet','Gerjan','Gentian','Gert','Valmir','Ylber','Ylli','Zaimin','Zamir','Zef','Zymbyl',
+
```
Comment (bram-pkg) [inline src/Faker/Provider/sq_AL/Person.php:46]: Spacing

File: src/Faker/Provider/sq_AL/Person.php [hunk 17]
```diff
@@ -0,0 +1,91 @@
...
+        'Klest','Klevis','Kodran','Krenar','Kreshnik','Krojan','Kujtim','Kushtim','Kushtrim','Labian','Laid','Laidan',
+        'Laidin','Lavdërim','Lavdim','Lavdimir','Lazrim','Lazror','Led','Ledian','Ledion','Lek','Lind','Lir','Liridon','Lirim','Lirjon','Luan',
+        'Luftar','Luftim','Lul','Lulash','Lulëzim','Llesh','Mal','Margjel','Marin','Mat','Mentar','Mentor','Migjen','Mir','Mirash','Mirjet',
+        'Mirjon','Mirush','Ndue','Neritan','Nertil','Nertilian','Ohri','Olt','Oltian','Oltion','Orges','Darli','Bruno','Pajtim','Pëllumb',
+        'Përparim','Petrit','Pirro','Pirust','Prend','Preng','Redon','Rron','Saimir','Skënder','Skerd','Sokol','Shkëlqim','Shkëlzen','Shkëmb',
+        'Shkumbin','Kristi','Kristjan','Shpend','Shpëtim','Tanush','Taulant','Teut','Tomë','Tomor','Tonin','Trëndafil','Trim','Tritan','Ukë','Ulqin',
+        'Urim','Val','Emilian','Emiliano','Valdet','Gerjan','Gentian','Gert','Valmir','Ylber','Ylli','Zaimin','Zamir','Zef','Zymbyl',
+        
+    ];
+    /**
```
Comment (bram-pkg) [inline src/Faker/Provider/sq_AL/Person.php:48]: Spacing

File: src/Faker/Provider/sq_AL/Person.php [hunk 18]
```diff
@@ -0,0 +1,91 @@
...
+        'Arba','Arbana','Arbena','Arbenore','Arberasha','Arbiana','Ardiana','Argjela','Arjeta','Arlinda','Armira','Arnisa','Artiana','Artina',
+        'Aulona','Bajame','Bardha','Bashkime','Buena','Besa','Besara','Besarta','Besiana','Besmira','Betime','Bukurie','Dafina','Dajtina','Dalina',
+        'Danja','Danjana','Dardana','Dardanesha','Dasantila','Dasara','Dashamira','Dava','Dëshira','Dija','Dila','Diljana','Durime','Dhurata',
+        'Dhurime','Edlira','Edrina','Egzona','Elira','Elona','Emira','Emiranda','Emiriana','Emirjona','Enkela','Enkelana','Entela','Era',
+        'Eranda','Erdita','Erëzake','Erina','Erinda','Erisa','Erza','Etleva','Ëngjëllushe','Fatbardha','Fatlume','Fatmira','Fitime','Fitore',
+        'Frangë','Garentina','Gazmira','Genca','Genciana','Genta','Gentiana','Gëzime','Gonxhe','Iliresha','Jasemina','Jehona','Jehora','Jela',
+        'Jerina','Jeta','Kastriote','Luliana','Luljeta','Olta','Oltiana','Ostrala','Pajtime','Panenta','Panjëz','Paqësime','Paqësore','Valbona',
+        'Vera','Verake','Veriana','Verime','Verina','Verjona','Verore','Veroshe','Verusha','Vesa','Vilma','Vitore','Vjollca','Vjosa','Xixa',
+        'Yllëza','Zamira','Zana',
+
```
Comment (bram-pkg) [inline src/Faker/Provider/sq_AL/Person.php:63]: Spacing

File: src/Faker/Provider/sq_AL/Address.php [hunk 19]
```diff
@@ -0,0 +1,69 @@
...
+        'Tish Daija', 'Avni Rustemi', 'Jordan Misja', 'Agim Nasufi','Naim Frashri', 'Migjeni', 'Pjeter Pograni',
+        'Agim Jashari', 'Isa Boletini','Pjeter Budi', 'Fran Vasa', 'Bajram Curri', 'Azem Galica', 'Albanopoli',
+        'Egnaita', 'Lek Dukagjini', 'Lefter Talo', 'Liri Gero', 'Lidhja e Prizrenit', 'Skernderbej', 'Kupa',
+        'Haki Taha', 'Faik Konica', 'Isuf Xhelili', 'Ali Asllani', 'Fato Berberi', 'Dardania', 'Princ Vidi',
+    ];
+    /**
+     * @see https://en.wikipedia.org/wiki/Foreign_relations_of_Albania
+     */
+    protected static $country = [
+        'Afganistan', 'Algjeria', 'American Samoa', 'Andorra', 'Angola','Antartide (territori a sud del 60° parallelo)', 'Argjentina', 'Armenia',
```
Comment (bram-pkg) [inline src/Faker/Provider/sq_AL/Address.php:31]: Spacing in this array is inconsistent.

**Pull request closed without merging**